pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -73,28 +74,32 @@ public Clients(final Client... clients) {
     }
 
     /**
-     * Initialize all clients by computing callback urls.
+     * Initialize all clients by computing callback urls if necessary.
      */
     @Override
     protected void internalInit() {
         CommonHelper.assertNotNull("clients", this.clients);
-        if (CommonHelper.isNotBlank(this.callbackUrl)) {
-            for (final Client client : this.clients) {
-                if (client instanceof IndirectClient) {
-                    final IndirectClient indirectClient = (IndirectClient) client;
-                    String indirectClientCallbackUrl = indirectClient.getCallbackUrl();
-                    // no callback url defined for the client -> set it with the group callback url
-                    if (indirectClientCallbackUrl == null) {
-                        indirectClient.setCallbackUrl(this.callbackUrl);
-                        indirectClientCallbackUrl = this.callbackUrl;
-                    }
-                    // if the "clientName" parameter is not already part of the callback url, add it unless the client
-                    // has indicated to not include it.
-                    if (indirectClient.isIncludeClientNameInCallbackUrl() &&
-                            indirectClientCallbackUrl.indexOf(this.clientNameParameter + "=") < 0) {
-                        indirectClient.setCallbackUrl(CommonHelper.addParameter(indirectClientCallbackUrl, this.clientNameParameter,
-                                indirectClient.getName()));
-                    }
+        final HashSet<String> names = new HashSet<>();
+        for (final Client client : this.clients) {
+            final String name = client.getName();
+            if (names.contains(name)) {
```

Comment:
I'm not sure to understand what you mean by " handled this logic inside the Clients instead". By default, client names are the classes names so if you have twice the same client, you will have the same name twice which would lead to unpredictable behaviours.
The idea here is to prevent that.